### PR TITLE
[WIP] Add template context tests

### PR DESCRIPTION
**Pull request by canyon289**
Anyone know how to actually use InferenceContext and if so can you post a snippet? With that I can update this PR

### CODE REVIEW

**Comment by ericmjl** [comment]
Darn this is some testing-fu! What's going on here, @canyon289? Can you also document the logic behind how you're writing these tests?

**Comment by canyon289** [comment]
> Darn this is some testing-fu! What's going on here, @canyon289? Can you also document the logic behind how you're writing these tests?

Yep, give me ~24 hours and I'll make these more clear. Thanks for the feedback

File: pymc4/tests/test_template_context.py [hunk 1]
```diff
@@ -27,7 +28,7 @@ def test_free_forward_context_add_variable(monkeypatch):
 
 
 def test_free_forward_context_var_as_backend_tensor(monkeypatch):
-    """Test that random variable initializes with in Free Forward context"""
+    """Test that random variable initialize in Free Forward context"""
```
Comment (ericmjl) [inline pymc4/tests/test_template_context.py]: Suggested grammatical change:

```
    """Test that random variable initializes in Free Forward context"""
```

Comment (canyon289) [reply pymc4/tests/test_template_context.py]: Let me know if you think the new docstring is good

**Comment by canyon289** [comment]
If no one has suggestions on how to use InferenceContext then this is ready for a merge

**Comment by twiecki** [comment]
Thanks @canyon289!

**Pull request merged**